backend/model/users.py

```diff
@@ -57,10 +57,21 @@ def delete_from_db(self):
         self.state = False
         db.session.commit()
 
-    def update_from_db(self, password):
+    def update_password_from_db(self, password):
         self.hash_password(password)
         db.session.commit()
 
+    def update_from_db(self, data):
+        if 0 < self.query.filter_by(username=self.username and id != data['id'], state=True).count():
+            raise Exception("Username already in use")
+        elif 0 < self.query.filter_by(email=self.email and id != data['id'], state=True).count():
```

Comment:
Similar al anterior.
`elif` innecesari perque si entra en la condicio anterior mai sortira.
```suggestion python
        if 0 < self.query.filter_by(email=data['email'] and id != self.id, state=True).count():
```